File: back-end/src/main/java/com/vecfonds/backend/service/ProductService.java
package com.vecfonds.backend.service;

import com.vecfonds.backend.entity.Image;
import com.vecfonds.backend.entity.Product;
import com.vecfonds.backend.payload.request.dto.ProductDTO;
import com.vecfonds.backend.payload.response.ProductResponse;
import org.springframework.stereotype.Service;
import org.springframework.web.multipart.MultipartFile;

import java.io.IOException;

@Service
public interface ProductService {
    ProductDTO convertProductDTO(Product product);
    ProductDTO createProduct(Long categoryId, Product product);
    ProductResponse getListProduct(Integer pageNumber, Integer pageSize, String sortBy, String sortOrder);
    ProductResponse getListProductByCategory(Integer pageNumber, Integer pageSize, String sortBy, String sortOrder, Long categoryId);
    ProductResponse getListProductByKeyword(Integer pageNumber, Integer pageSize, String sortBy, String sortOrder, String keyword);
    ProductDTO updateProduct(Long productId, Product product);
    String deleteProduct(Long productId);
    ProductDTO createProductImageLink(Long productId,Image image);
    ProductDTO createProductImageMultipartFile(Long productId, MultipartFile image, Integer main) throws IOException;
}

File: front-end/src/services/product.service.js
import axios from "axios";

const API_URL = `${process.env.REACT_APP_SERVER_URL}/product`;

const getProduct = ({ productId }) => {
  return axios.get(`${API_URL}/${productId}`,
    {
      headers: {
        "Content-Type": "application/json",
      }
    });
};

const getListProduct = ({ pageNumber, pageSize, sortBy, sortOrder }) => {
  return axios.get(`${API_URL}/list?pageNumber=${pageNumber}&pageSize=${pageSize}&sortBy=${sortBy}&sortOrder=${sortOrder}`,
    {
      headers: {
        "Content-Type": "application/json",
      }
    });
};

const getListProductByCategory = ({ pageNumber, pageSize, sortBy, sortOrder, categoryId }) => {
  return axios.get(`${API_URL}/list/category/${categoryId}?pageNumber=${pageNumber}&pageSize=${pageSize}&sortBy=${sortBy}&sortOrder=${sortOrder}`,
    {
      headers: {
        "Content-Type": "application/json",
      }
    });
};

const getListProductByKeyword = ({ pageNumber, pageSize, sortBy, sortOrder, keyword }) => {
  return axios.get(`${API_URL}/list/keyword/${keyword}?pageNumber=${pageNumber}&pageSize=${pageSize}&sortBy=${sortBy}&sortOrder=${sortOrder}`,
    {
      headers: {
        "Content-Type": "application/json",
      }
    });
};


const productService = {
  getProduct,
  getListProduct,
  getListProductByCategory,
  getListProductByKeyword
};

export default productService;
